test/integration/rememberable_test.rb

```diff
@@ -127,37 +127,83 @@ def cookie_expires(key)
     end
   end
 
-  test 'extends remember period when extend remember period config is true' do
```

Comment:
The tests added in https://github.com/heartcombo/devise/pull/4039 were incorrect; [this assertion](https://github.com/heartcombo/devise/pull/4039/files#diff-902720baf2c3ec018ed1e9f3d8829044515573eea37c8aee6234d9b4e2ef02c7R135) was passing because the user was getting logged out (because the test moves backwards in time). Added an `assert_response` to confirm this, then refactored the test to be correct.